compliance/kibana.go

```diff
@@ -380,35 +384,13 @@ func (k *Kibana) MustExistDashboard(dashboardID string) error {
 }
 
 func (k *Kibana) getDashboard(dashboardID string) (*dashboardResponse, error) {
-	// Before it was used /api/dashboards/dashboard/{dashboardID} endpoint but it is in
-	// technical preview and it is not available in all Kibana versions by default.
-	// Related to https://github.com/elastic/kibana/pull/223262
-	// Use instead Saved Objects API to get the dashboard.
-	apiPath, err := url.JoinPath(apiGetDashboardPath, dashboardID)
-	if err != nil {
-		return nil, err
-	}
-	log.Println("Using API path:", apiPath)
-
-	req, err := k.newRequest(http.MethodGet, apiPath, nil)
+	// Use the export API to retrieve the given dashboard given its "id".
+	body := fmt.Sprintf(`{"objects": [{"id": "%s", "type": "dashboard"}],"excludeExportDetails":true,"includeReferencesDeep":false}`, dashboardID)
+	req, err := k.newRequest(http.MethodPost, apiExportSavedObjectPath, strings.NewReader(body))
```

Comment:
Interestingly this request is [documented](https://www.elastic.co/docs/api/doc/kibana/v8/operation/operation-exportsavedobjectsdefault) to only return 200 or 400 errors, and it returns HTTP 400 status code when the saved object doesn't exist, but it wraps a 404 error inside:
```
request failed with status 400, body: {"statusCode":400,"error":"Bad Request","message":"Error fetching objects to export","attributes":{"objects":[{"id":"basic_content-dashboard-xyz-1","type":"dashboard","error":{"statusCode":404,"error":"Not Found","message":"Saved object [dashboard/basic_content-dashboard-xyz-1] not found"}}]}}
```

Well, it is probably because it returns a code for each requested object.